scripts/get_distance_to_focal_set.py

```diff
@@ -0,0 +1,183 @@
+        val[n_snps:right] = s[snps]
+        r += 1
+        n_snps = right
+        if chunk_size and chunk_size==nseqs:
+            break
+
+    if nseqs==0:
+        return None
+
+    row = row[0:right]
+    col = col[0:right]
+    val = val[0:right]
+
+    sparse_snps = sparse.csc_matrix((val, (row, col)), shape=(nseqs, align_length))
+
+    return {'snps': sparse_snps, 'consensus': consensus, 'names': seq_names, 'filled_positions': filled_positions}
+
+# Function adapted from https://github.com/gtonkinhill/pairsnp-python
+def calculate_distance_matrix(sparse_matrix_A, sparse_matrix_B, consensus):
+
+    n_seqs_A = sparse_matrix_A.shape[0]
+    n_seqs_B = sparse_matrix_B.shape[0]
+
+    d = (1*(sparse_matrix_A==97)) * (sparse_matrix_B.transpose()==97)
+    d = d + (1*(sparse_matrix_A==99) * (sparse_matrix_B.transpose()==99))
+    d = d + (1*(sparse_matrix_A==103) * (sparse_matrix_B.transpose()==103))
+    d = d + (1*(sparse_matrix_A==116) * (sparse_matrix_B.transpose()==116))
+
+    d = d.todense()
+
+    n_comp = (1*(sparse_matrix_A==110) * ((sparse_matrix_B==110).transpose())).todense()
+    d = d + n_comp
+
+    temp_total = np.zeros((n_seqs_A, n_seqs_B))
+    temp_total[:] = (1*(sparse_matrix_A>0)).sum(1)
+    temp_total += (1*(sparse_matrix_B>0)).sum(1).transpose()
+
+    total_differences_shared = (1*(sparse_matrix_A>0)) * (sparse_matrix_B.transpose()>0)
+
+    n_total = np.zeros((n_seqs_A, n_seqs_B))
+    n_sum = (1*(sparse_matrix_A==110)).sum(1)
+    n_total[:] = n_sum
+    n_total += (1*(sparse_matrix_B==110)).sum(1).transpose()
+
+    diff_n = n_total - 2*n_comp
+    d = temp_total - total_differences_shared.todense() - d - diff_n
+
+    return d
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="generate priorities files based on genetic proximity to focal sample",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("--alignment", type=str, required=True, help="FASTA file of alignment")
+    parser.add_argument("--reference", type = str, required=True, help="reference sequence (fasta)")
+    parser.add_argument("--focal-alignment", type = str, required=True, help="focal smaple of sequences")
+    parser.add_argument("--output", type=str, required=True, help="FASTA file of output alignment")
+    args = parser.parse_args()
+
+    # load entire alignment and the alignment of focal sequences (upper case -- probably not necessary)
+    ref = sequence_to_int_array(SeqIO.read(args.reference, 'fasta').seq)
+    alignment_length = len(ref)
+
+    fh_focal = open(args.focal_alignment, 'rt')
+    focal_seqs = SimpleFastaParser(fh_focal)
+    focal_seqs_dict = calculate_snp_matrix(focal_seqs, consensus = ref)
+
+    fh_seqs = open(args.alignment, 'rt')
+    seqs = SimpleFastaParser(fh_seqs)
+
+    # export priorities
+    fh_out = open(args.output, 'w')
+    fh_out.write('strain\tclosest strain\tdistance\n')
+
+    chunk_size=10000
```

Comment:
This is excellent! Would parameterizing this variable in the script's command line arguments effectively allow users to tune memory required by this script? That could be helpful for folks who may have different memory constraints.

Reply:
indeed. this should be a parameter. I don't think there much of a performance cost if this parameter is 1000 or larger. but it will start requiring unreasonable amounts of memory if it is in excess of 100k. In between, I wouldn't expect this to matter much.